pkgs/by-name/py/pyrefly/package.nix

```diff
@@ -8,23 +8,33 @@
 }:
 rustPlatform.buildRustPackage (finalAttrs: {
   pname = "pyrefly";
-  version = "0.34.0";
+  version = "0.42.2";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "pyrefly";
     tag = finalAttrs.version;
-    hash = "sha256-HPPDsvWEFfh/GNMUPiVjQr28YBBs2DACBGM3cxo5Nx4=";
+    hash = "sha256-jNQnqiIZHSAzL/6m7AGxGuhFvlOZZLATbm5MX9aReaw=";
   };
 
+  patches = [
+    ./notebook_test.patch
+    ./shebang.patch
+  ];
+
   buildAndTestSubdir = "pyrefly";
-  cargoHash = "sha256-46kcoBG/PWwf8VdlvLNzEhfYRTmmKi/uTjwFkl7Wozg=";
+  cargoHash = "sha256-nOVMsnaKYAHDVt/JjMsGpj+ZKWBebbMV0iTf3pdYAPA=";
 
   buildInputs = [ rust-jemalloc-sys ];
 
   nativeInstallCheckInputs = [ versionCheckHook ];
   doInstallCheck = true;
 
+  # redirect tests writing to /tmp
+  preCheck = ''
+    export TMPDIR=$(mktemp -d)
+  '';
+
```

Comment:
```suggestion
```
Suggestion: add `writableTmpDirAsHomeHook` to `nativeCheckInputs` instead.

Reply:
@GaetanLepage The tests try to write to `/tmp/`. Would making `$HOME` writable help with that? Unless I'm misunderstanding the docs here: https://nixos.org/manual/nixpkgs/stable/#writable-tmpdir-as-home.sh

Upstream said `TMPDIR` needs to be set to something writeable.